core/src/main/java/org/apache/accumulo/core/clientImpl/InstanceOperationsImpl.java

```diff
@@ -119,6 +127,38 @@ public void modifyProperties(final Consumer<Map<String,String>> mapMutator)
     // Send to server
     ThriftClientTypes.MANAGER.executeVoid(context, client -> client
         .modifySystemProperties(TraceUtil.traceInfo(), context.rpcCreds(), vProperties));
+
+    return vProperties.getProperties();
+  }
+
+  @Override
+  public Map<String,String> modifyProperties(final Consumer<Map<String,String>> mapMutator)
+      throws AccumuloException, AccumuloSecurityException, IllegalArgumentException {
+
+    var log = LoggerFactory.getLogger(InstanceOperationsImpl.class);
+
+    Retry retry =
+        Retry.builder().infiniteRetries().retryAfter(25, MILLISECONDS).incrementBy(25, MILLISECONDS)
+            .maxWait(30, SECONDS).backOffFactor(1.5).logInterval(3, MINUTES).createRetry();
+
+    while (true) {
+      try {
+        var props = tryToModifyProperties(mapMutator);
+        retry.logCompletion(log, "Modifying instance properties");
+        return props;
+      } catch (ConcurrentModificationException cme) {
+        try {
+          retry.logRetry(log,
+              "Unable to modify instance properties for because of concurrent modification");
+          retry.waitForNextAttempt();
+        } catch (InterruptedException e) {
+          throw new RuntimeException(e);
+        }
+      } finally {
+        retry.useRetry();
```

Comment:
I wonder if instead of calling the method `Retry.logCompletion`, it should be called `Retry.completed(msg)` and then it disables any checks that might throw errors. For example:

I don't think this is a problem here, because this Retry is configured for infiniteRetries. However, if Retry.useRetry is called elsewhere in a finally block, it could throw an exception even if the operation completed successfully (in the case where the Retry was configured for N retries, and it had retried N times but succeeding on the last one.

Reply:
Looking further, it looks like the issue can be avoided, if in a finally block (and maybe elsewhere where useRetry is called:

```
   ...
   } finally {
     if (retry.canRetry()) {
      retry.useRetry();
     }
  }
```

Reply:
Using the Rety class I find myself wondering if it can be improved and then more widely used in the code. Like maybe pass an interface to a method that will do the retries, looping, logging etc.  The interface can supply the operation to run and the messages to log.

Reply:
Continuation or prev comment.  Wondering about something like the following.  Thinking through it a bit i think dealing with checked exceptions would be the biggest pain point in doing something like this.

```java
Retry retry = ..
retry.execute(new Retriable(){
    //whatever is needed to execute a retriable operation : Runnable, descriptive message, etc
});
```

Reply:
I did something similar when we were working on external compactions to retry thift calls - [RetryableThriftCall](https://github.com/apache/accumulo/blob/main/server/base/src/main/java/org/apache/accumulo/server/compaction/RetryableThriftCall.java)

Reply:
> I did something similar when we were working on external compactions to retry thift calls - [RetryableThriftCall](https://github.com/apache/accumulo/blob/main/server/base/src/main/java/org/apache/accumulo/server/compaction/RetryableThriftCall.java)

We need something more general like that so we can reuse retry code.  Still scratching my head over the best way to handle checked exceptions though.  Like what the most concise way to handle one operation that throws checked exception X and another that throws checked exception Y and these are checked exceptions that we want to pass up.

Reply:
I think this is a case that's similar to `Future.get()` where it throws an `ExecutionException`. As an example, we could:

1. modify  [RetryableThriftFunction](https://github.com/apache/accumulo/blob/main/server/base/src/main/java/org/apache/accumulo/server/compaction/RetryableThriftFunction.java) to declare `Exception` rather than `TException'. 
2. Then in [RetryableThrifCall](https://github.com/apache/accumulo/blob/main/server/base/src/main/java/org/apache/accumulo/server/compaction/RetryableThriftCall.java#L106) catch `Exception`
3. When retries have been exhausted, throw a RetriesExceededException [here](https://github.com/apache/accumulo/blob/main/server/base/src/main/java/org/apache/accumulo/server/compaction/RetryableThriftCall.java#L118) when there is no error and throw an ExecutionException when retries have been exhausted and there is an error.

Reply:
@dlmarion I think it would be good to open up some sort of follow on issue about improving retry, but not sure what exactly.  I think you have more defined thoughts on it.